src/eureka/lib/naninterp1d.py

```diff
@@ -0,0 +1,51 @@
+# Interpolate over NaNs in 1D
+
+import numpy as np
+
+
+def naninterp1d(y, replace_val=0):
+    """
+    Interpolate over NaNs in 1D array.
+
+    Parameters
+    ----------
+    y : 1D array
+        Data array with possible NaNs
+    replace_val : float, optional
+        Value to use when entire dataset is NaNs (default is 0).
+
+    Returns
+    -------
+    y : 1D array
+        Data array without NaNs
+    """
+    nans, x = nan_helper(y)
+    if len(y[~nans]) > 0:
+        y[nans] = np.interp(x(nans), x(~nans), y[~nans])
+    else:
+        y[nans] = replace_val
+    return y
+
+
+def nan_helper(y):
```

Comment:
This doesn't really feel like it needs to be a separate function to me, but we can leave it as one if you want. I'd likely just do the following up above:
```
nans = np.isnan(y)
x = lambda z: z.nonzero()[0]
```

Reply:
Looking more closely now, there's also no need to define the x lambda function. I think a simpler `naninterp1d` function would be:
```
nans = np.isnan(y)
if np.all(nans):
    y[nans] = replace_val
else:
    y[nans] = np.interp(np.where(nans)[0], np.where(~nans)[0], y[~nans])
return y
```

I'm not 100% sure you need the `[0]`'s above, but I think so